### PR TITLE
added function to get visit label to instrument class

### PR DESCRIPTION
**Pull request by jesscall**
This pull request adds a function to get the visit label of a given timepoint. It will be useful in instances where certain instrument information is different (ex:headers) for different visits.

### CODE REVIEW

File: php/libraries/NDB_BVL_Instrument.class.inc [hunk 1]
```diff
@@ -1398,6 +1398,22 @@ class NDB_BVL_Instrument extends NDB_Page
         return false;
     }
 
+    /**
+     * Gets the VisitLabel for the timepoint to which this instrument pertains
+     *
+     * @return String for the VisitLabel
+     */
+    function getVisitLabel{
```
Comment (driusan) [inline php/libraries/NDB_BVL_Instrument.class.inc]: this isn't syntactically valid PHP. You need the argument list.

(You should also run phpcs on it or Travis is going to fail. You can just run `npm run lint:php` as a shortcut to run PHPCS and `php -l`.)

File: php/libraries/NDB_BVL_Instrument.class.inc [hunk 2]
```diff
@@ -1398,6 +1398,22 @@ class NDB_BVL_Instrument extends NDB_Page
...
+        $sessionID = $this->getSessionID();
+
+        if(!empty($sessionID)){
+            $t = Timepoint::singleton($sessionID);
+            $visitLabel = $t->getVisitLabel();
+            return $visitLabel;
+        }
+        return false;
```
Comment (driusan) [inline php/libraries/NDB_BVL_Instrument.class.inc]: false isn't a string

**Comment by um4r12** [comment]
LGTM

File: php/libraries/NDB_BVL_Instrument.class.inc [hunk 3]
```diff
@@ -1398,6 +1398,23 @@ class NDB_BVL_Instrument extends NDB_Page
...
+     * Gets the VisitLabel for the timepoint to which this instrument pertains
+     *
+     * @return String for the VisitLabel
+     */
+    function getVisitLabel()
+    {
+        $sessionID = $this->getSessionID();
+
+        if (!empty($sessionID)) {
+            $t          = Timepoint::singleton($sessionID);
```
Comment (zaliqarosli) [inline php/libraries/NDB_BVL_Instrument.class.inc]: i am wondering if this should be namespaced? i.e. \Timepoint::singleton($sessionID)?

Comment (zaliqarosli) [inline php/libraries/NDB_BVL_Instrument.class.inc]: i am wondering if this should be namespaced? i.e. \Timepoint::singleton($sessionID)?

**Comment by jesscall** [comment]
@kongtiaowang Can you help with this? not sure why travis is failing

**Comment by kongtiaowang** [comment]
Travis is back!

**Comment by driusan** [comment]
Thinking on this a little more, I'm a little opposed to this because I think it would be better to have something that returns a typed visit object rather than a string. Adding this to minor would encourage people to use this without the safety of type checking.

On the other hand, the "Visit" object is part of #3579 and I don't know what the timeline on that getting merged is.

**Comment by driusan** [comment]
@xlecours any thoughts on the above comment?

**Comment by PapillonMcGill** [comment]
@driusan @jesscall @xlecours 
the getVisitLabel in the TimePoint object already return a string, so this just extend on it. 
Of course it would be better to return an object as we will be able to use the getName() function of the object.

As for the [3579](https://github.com/aces/Loris/pull/3579/files), as you know my schedule is a little bit upside down. Can't promise anything now but hope soon. Worse case, we could modify all use of visit afterwards.

**Comment by driusan** [comment]
There doesn't seem to be any movement on the other one, so I'll merge this, but we should deprecate this once we have an object to use instead.

**Pull request merged**